[AArch64] Support MI and PL

AZero13 opened the pull request:
Now, why would we want to do this?

There are a small number of places where this works:
1. It helps peepholeopt when less flag checking.
2. It allows the folding of things such as x - 0x80000000 < 0 to be folded to cmp x, register holding this value
3. We can refine the other passes over time for this.

AZero13 commented:
@davemgreen thoughts on this?

davemgreen commented on llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:
```diff
@@ -3320,7 +3320,8 @@ static bool isZerosVector(const SDNode *N) {
 /// changeIntCCToAArch64CC - Convert a DAG integer condition code to an AArch64
 /// CC
-static AArch64CC::CondCode changeIntCCToAArch64CC(ISD::CondCode CC) {
+static AArch64CC::CondCode changeIntCCToAArch64CC(ISD::CondCode CC, SDValue RHS,
```
Could this use SDValue RHS = {} to not require canUsePLOrMI?

davemgreen reviewed (commented):
I think that simpler flags is a nice thing to have.

AZero13 committed:
[AArch64] Support MI and PL

Now, why would we want to do this?

There are a small number of places where this works:
1. It helps peepholeopt when less flag checking.
2. It allows the folding of things such as x - 0x80000000 < 0 to be folded to cmp x, register holding this value
3. We can refine the other passes over time for this.

davemgreen commented on llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:
```diff
@@ -4065,7 +4066,7 @@ static SDValue getAArch64Cmp(SDValue LHS, SDValue RHS, ISD::CondCode CC,
             SExt, DAG.getSignedConstant(ValueofRHS, DL, RHS.getValueType()), CC,
             DL, DAG);
-        AArch64CC = changeIntCCToAArch64CC(CC);
+        AArch64CC = changeIntCCToAArch64CC(CC, RHS);
```
This can only be < 0?

davemgreen reviewed (commented):
Thanks. LGTM other than the points below.

AZero13 commented:
> Thanks. LGTM other than the points below.

Fixed!

AZero13 commented:
Rebased because I had to resolve conflicts.

AZero13 commented:
@davemgreen done!
